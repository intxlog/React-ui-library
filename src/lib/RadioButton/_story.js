import React from 'react'
import { storiesOf } from '@storybook/react'
import Input from './../Input'

storiesOf('Radio Button', module)
.add('default', () => (
  <Input
    type='radio'
    name={'test'}
    value={`option1`}
    idForLabel={`option1`}
    labelText={'Option 1'}
  ></Input>
))
.add('defaultChecked', () => (
  <Input
    type='radio'
    name={'test'}
    value={`option1`}
    idForLabel={`option1`}
    labelText={'Option 1'}
    defaultChecked
  ></Input>
))
.add('disabled', () => (
  <Input
    type='radio'
    name={'test'}
    value={`option1`}
    idForLabel={`option1`}
    labelText={'Option 1'}
    disabled
  ></Input>
))