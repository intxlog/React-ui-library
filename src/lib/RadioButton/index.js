import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

//import css
import styles from './styles.module.scss'

const RadioButton = ({
  id,
  value,
  name,
  defaultChecked,
  disabled,
  labelText,
  error,
  onChange,
  onBlur
}) => {
  return (
    <div className={classNames({
      [styles.container]: true,
      [styles.error]: error,
      [styles.disabled]: disabled
    })}>
      <input
        id={id}
        value={value}
        name={name}
        type={`radio`}
        defaultChecked={defaultChecked}
        disabled={disabled}
        onChange={onChange}
        onBlur={onBlur}
      />
      <div className={styles.element}></div>
      <label
        htmlFor={id}
      >{labelText}</label>
    </div>
  )
}

//rules for props being passed in
RadioButton.propTypes = {
  id: PropTypes.string,
  value: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.bool
  ]).isRequired,
  name: PropTypes.string,
  defaultChecked: PropTypes.bool,
  disabled: PropTypes.bool,
  labelText: PropTypes.string.isRequired,
  error: PropTypes.bool,
  onChange: PropTypes.func,
  onBlur: PropTypes.func
}

//maintain the name for documentation purposes
RadioButton.displayName = 'RadioButton'

export default RadioButton