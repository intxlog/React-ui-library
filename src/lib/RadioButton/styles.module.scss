@import './../../styles/_helpers.scss';

$heightWidth: 14px;
$circleHeightWidth: 8px;
$circleOffset: 2px;

.container {
  position: relative;
  box-sizing: border-box;

  > * {
    box-sizing: border-box;
  }

  > label {
    font-size: 16px;
    line-height: 22px;
    cursor: pointer;
    padding-left: 18px;
    display: block;
  } 
  
  > input {
    font-family: inherit;
    font-size: 100%;
    line-height: 1.15;
    margin: 0;
    overflow: visible;
    padding: 0;
    clip: rect(1px, 1px, 1px, 1px);
    width: 1px;
    height: 1px;
    position: absolute;
    
    &:checked ~ .element::before {
      display: block;
    } 
  }
  
  > .element {
    background-color: white;
    border-radius: 100%;
    height: $heightWidth;
    width: $heightWidth;
    cursor: pointer;
    border: 1px solid $colorFormInputBorder;
    position: absolute;
    top: 4px;
    pointer-events: none;
    
    &:before {
      content: '';
      background-color: $colorFormInputBorder;
      height: $circleHeightWidth;
      width: $circleHeightWidth;
      position: absolute;
      top: $circleOffset;
      left: $circleOffset;
      border-radius: 100%;
      display: none;
    }
  }

  &.disabled {
    > label {
      opacity: .5;
    }

    > .element {
      background-color: $colorFormInputDisabledBackground;
    }
  }

  &.error {
    > label {
      color: red;
    }

    > .element {
      background-color: rgb(249, 154, 154);
      border: red 1px solid;
    }
  }
}