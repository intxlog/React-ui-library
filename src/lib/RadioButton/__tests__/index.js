import React from 'react'
import { shallow } from 'enzyme'
import RadioButton from '..'

describe('RadioButton', () => {
  let props = {
    labelText: `Example`,
    value: true
  }
  
  //create a shallow wrapper that can be reused
  let wrapper
  //initialize the components
  beforeEach(() => {
    wrapper = shallow(<RadioButton {...props}/>)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })
})