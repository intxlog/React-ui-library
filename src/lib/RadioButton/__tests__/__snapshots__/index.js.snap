// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioButton renders correctly 1`] = `
<div
  className="container"
>
  <input
    type="radio"
    value={true}
  />
  <div
    className="element"
  />
  <label>
    Example
  </label>
</div>
`;
